types/netmap/netmap.go

```diff
@@ -252,6 +252,22 @@ func (nm *NetworkMap) DomainName() string {
 	return nm.Domain
 }
 
+// TailnetDisplayName returns the admin-editable name contained in
+// NodeAttrTailnetDisplayName. If the capability is not present it
+// returns an empty string.
+func (nm *NetworkMap) TailnetDisplayName() string {
+	if nm == nil || !nm.HasCap(tailcfg.NodeAttrTailnetDisplayName) {
+		return ""
+	}
+
+	tailnetDisplayNames, err := tailcfg.UnmarshalNodeCapJSON[string](nm.SelfNode.CapMap().AsMap(), tailcfg.NodeAttrTailnetDisplayName)
```

Comment:
not sure if `CapMap().AsMap()` was the correct thing to use here @nickkhyl, I was mostly relying on type checking to find the right thing as other usage of `UnmarshalNodeCapJSON` have a `Node` available to them.